source/cloud-native-platform/health-end-points/index.html.md.erb

```diff
@@ -0,0 +1,30 @@
+---
+title: Health Probes
+last_reviewed_on: 2023-12-12
+review_in: 6 months
+weight: 90
+---
+
+# <%= current_page.data.title %>
+
+## Readiness Checks on Kubernetes
+
+Readiness probes are crucial in Kubernetes as they are called periodically, impacting container availability. 
+Failing readiness checks make containers temporarily unavailable, preventing incoming requests from reaching your application. It's essential to consider what checks to include in readiness probes to avoid unnecessary downtime.
+
+### Considerations:
+
+- Include exclusive/hard dependencies for your service in readiness checks.
+- Avoid including soft dependencies; handle their unavailability in code for a better customer experience.
+
+**Good Example:**
+- A private cache/database (e.g., Redis or Postgres) exclusive to the application.
+
+**Bad Example:**
+- Shared components like IDAM, S2S, or CCD.
+
+## Add Database Check to Liveness Probe
```

Comment:
The more i think about it, i am getting a bit worried about adding DB checks to liveness. Especially given the app will keep crashing till its resolved. This might cause some pressure on node if its not self recoverable. So may be, better leave it in readiness.

Reply:
but to automatically resolve the issue the app needs to restart, if we put it in readiness it will never self resolve unfortunately =/.

Reply:
i guess we could test it a bit more with readiness checks, break the networking. Make a few calls and see it failing, then restore it, and see if it recovers?

Reply:
yeah, needs some testing and yeah recovering will depend on the nature of issue as well.

Reply:
Tested db check in readiness probe by breaking this line https://github.com/hmcts/cnp-plum-recipes-service/blob/DTSPO-14035-Testing-db-check-on-readiness1/src/main/resources/application.yaml#L11
check came back with 404,  restored the line and the db check came back with a status up.

Reply:
On Friday,  tested with shutting down the db server,  this cause the pod to crash.  started the db server up again and the pod restored fine and was able to hit the db check health endpoint.

Reply:
was this with readiness and liveness probes?

Reply:
This was with the readiness probe.

Reply:
ok can you update this with readiness then please:
https://github.com/hmcts/spring-boot-template/blob/master/src/main/resources/application.yaml#L10

lets go with that for now.